### PR TITLE
PR:show thumbs when leadimage behavior enabled for eventtype:

### PR DESCRIPTION
**Pull request by fgrcon**
see  https://github.com/plone/Products.CMFPlone/issues/1226

### CODE REVIEW

**Comment by jensens** [comment]
@thet may you please review?

**Comment by thet** [comment]
Hi, tnx for the pull request, and especially your work on the content type icon issues which was awesome!

But I have some problems with this pull request:

- The LeadImage preview thumbnails in listings is a new feature and should have gone through a discussion process in the @plone/framework-team - I even think a PLIP would have been needed.
- The pull-requests regarding the content-type issue and LeadImage thumbnails were done together - as far as I know. Which removes the possibility of accepting the content-type icon bugfix and the leadimage thumb feature separately.
- All the markup gets more complicated (actually my main concern).
- The feature could lead to inconsistent behavior, where potentially not all listings implement the LeadImage thumb preview - especially addon listings.
- The feature is immediately active for all news items which uses the LeadImage behavior without a chance to remove the thumbs.
- A registry option to deactivate the feature is missing.
- The layout of the LeadImage preview icons is not optimal at the moment.
    - The link-hover-underline line is also below the image, which doesn't look very nice.
    - The image alignment isn't optimal. Maybe it should be float to the right? @plone/ux-team 

Before this is merged, I'd like to further discuss this issue: /cc @vangheem @plone/framework-team 

Sorry about complaining so late in the process, where you already invested a lot of work! I hope I don't put too much stop energy on to your Plone development enthusiasm! But for the next time, please stick to the following rules described in these documents:

http://docs.plone.org/develop/coredev/docs/index.html
http://docs.plone.org/develop/coredev/docs/plips.html

**Comment by thet** [comment]
Here is a screenshot, how it looks currently for events:
screen shot 2015-11-13 at 11 03 09

**Comment by fgrcon** [comment]
:-) hi,

If I'd known the hornets nest, I stirred up with the proposal  for a small enhancement (https://github.com/plone/Products.CMFPlone/issues/1074) **I most likely  would never have gotten involved in that stuff.**

Digging deeper an deeper i found quite a number of things: 
* old Icons / new icons
* lots of dead code (?)
* missing specifications (or I just dit not find them ...) 
* https://github.com/plone/Products.CMFPlone/issues/1185
* quite new: 
....

*  **what I tried, was simply to clean up and standardize a little.**  
* In https://github.com/plone/plone.app.content/pull/57 I agreed to a plip  ....

to get a better overview I cleaned up issues and pr s -> https://github.com/plone/Products.CMFPlone/issues/1226
inluding a link to https://github.com/fgrcon/Products.CMFPlone/blob/documented/docs/icons_in_plone5.rst
which covers some of your complaints.

e.g.:
 supress thumbnails:  
````
  .image-icon, .image-tile {display:none} 
````
you can do that on a global or detailed scope ...

I just did not find out how to control that from the site-controlpanel. Maybe you can help.

Mark up did not get much more complicated, it tried even  to clean up some.

optimizing css is a topic but that allways seems to be 


It was quite a lot of work but I  learned a lot about plone and the stuff around (especially the processes), and I suppose i will get a degree in software archeology soon. And if you want to know:
You did neither stop nor annoy me (i am a "steirischer Sturschädel" :).

I will keep to the standards and maybe I also will come up with some proposals concerning the processes ( that s just what I was working on for more than 25 years).

**Comment by fgrcon** [comment]
https://github.com/plone/plonetheme.barceloneta/pull/60:

events-portlet

**Comment by thet** [comment]
oh - nice! looks much better.
i have to run now... maybe others can join in, otherwise i'll come back to it on monday.

File: plone/app/event/portlets/portlet_events.py [hunk 1]
```diff
@@ -182,7 +182,11 @@ def events(self):
                 ret_mode=RET_MODE_ACCESSORS,
                 expand=True, limit=data.count, **query
             )
-        return events
+            eventlist=[]
+            for ev in events:
+                hasimage = True if getattr(ev.context, 'image', None) else False
```
Comment (jensens) [inline plone/app/event/portlets/portlet_events.py]: or ``bool(getattr(ev.context, 'image', False))`` ;) (easier to read)

Comment (jensens) [inline plone/app/event/portlets/portlet_events.py]: easier to read:  ``bool(getattr(ev.context, 'image', False))``

Comment (jensens) [inline plone/app/event/portlets/portlet_events.py]: btw, this changes it only for the second (else) part if the ICollection condition above. I assume this wont work with collections.

**Comment by jensens** [comment]
test anyway http://jenkins.plone.org/job/pull-request-5.0/545/

**Comment by thet** [comment]
i refactored this pull a bit, so that it doesn't break code, when the event_portlet.pt is overloaded by integration products via z3c.jbot. see: https://github.com/plone/plone.app.event/commit/7aa0f0227a73efbba40668b7fc6fdd3dfa1c6503

@fgrcon are you a "steirischer sturschädl" living in vienna or styria? i'm also one, living in graz. if graz, we could make a plone meetup once.

**Comment by fgrcon** [comment]
I live in Vienna but graz is not that far. I sent a Xing invitation rright now.

**Comment by jensens** [comment]
@fgrcon Austrians all over the place ;) I'am living in Innsbruck - so what about a beer next time I'am in Vienna with a time-slot left?

**Comment by fgrcon** [comment]
@jensens :+1:

**Pull request merged**